test/hotspot/jtreg/gc/shenandoah/TestShenandoahRegionLogging.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test id=rotation
+ * @requires vm.gc.Shenandoah
+ *
+ * @run main/othervm -Xmx1g -Xms1g -XX:+UnlockDiagnosticVMOptions -XX:+UnlockExperimentalVMOptions
+ *      -XX:+ShenandoahRegionSampling
+ *      -Xlog:gc+region=trace:region-snapshots-%p.log::filesize=100,filecount=3
+ *      -XX:+UseShenandoahGC
+ *      TestShenandoahRegionLogging
+ */
+import java.io.File;
+import java.util.Arrays;
+import java.nio.file.Files;
+
+public class TestShenandoahRegionLogging {
+    private static final long TARGET_MB = Long.getLong("target", 1);
+    private static Object sink;
+
+    public static void main(String[] args) throws Exception {
+        long count = TARGET_MB * 1024 * 1024 / 16;
+        for (long c = 0; c < count; c++) {
+            sink = new Object();
+            Thread.sleep(1);
```

Comment:
Pre-existing, but: The problem with doing this sleep is that we are doing about 64K of 1ms sleeps in this tests. Even if we manage to do 1ms sleep, that is already 65 seconds spent in this test. But we might as well spend single-digit ms per sleep. For 10ms sleeps, that is 10 minutes for a test, which would be playing it close to timeout.

So I wonder if this test should be time-bound.

Reply:
Would it be better to reduce the frequency of the sleep, to one in 10, for example?

Reply:
The test is really just waiting for region sampling to write to the logs. I think one call to `System.gc` would suffice. I'll see if I can reduce the runtime for this test.